internal/service/handlers/create_balance_v2.go

```diff
@@ -0,0 +1,58 @@
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/rarimo/geo-auth-svc/pkg/auth"
+	"github.com/rarimo/geo-points-svc/internal/data"
+	"github.com/rarimo/geo-points-svc/internal/service/requests"
+	"gitlab.com/distributed_lab/ape"
+	"gitlab.com/distributed_lab/ape/problems"
+)
+
+func CreateBalanceV2(w http.ResponseWriter, r *http.Request) {
+	req, err := requests.NewCreateBalanceV2(r)
+	if err != nil {
+		ape.RenderErr(w, problems.BadRequest(err)...)
+		return
+	}
+
+	nullifier := req.Data.ID
+	if !auth.Authenticates(UserClaims(r), auth.UserGrant(nullifier)) {
+		ape.RenderErr(w, problems.Unauthorized())
+		return
+	}
+
+	balance, err := BalancesQ(r).FilterByNullifier(nullifier).Get()
+	if err != nil {
+		Log(r).WithError(err).Error("Failed to get balance by nullifier")
+		ape.RenderErr(w, problems.InternalError())
+		return
+	}
+
+	if balance != nil {
+		ape.RenderErr(w, problems.Conflict())
+		return
+	}
+
+	err = EventsQ(r).Transaction(func() error {
+		events := prepareEventsWithRef(nullifier, "", false, r)
```

Comment:
Bug: new event in config and probably in admin console won't be added for disabled balances, but you aren't adding it on activation. Solutions:
1) Rollback changes in `reopener` and admin endpoints to open events for disabled balances
2) Move logic of reopening into balance activation

Reply:
solution 1 chosen